packages/nuxt/src/runtime/hooks/updateRouteBeforeResponse.ts

```diff
@@ -0,0 +1,49 @@
+import { getActiveSpan, getCurrentScope, getRootSpan, logger, SEMANTIC_ATTRIBUTE_SENTRY_SOURCE } from '@sentry/core';
+import type { H3Event } from 'h3';
+
+/**
+ * Update the root span (transaction) name for routes with parameters based on the matched route.
+ */
+export function updateRouteBeforeResponse(event: H3Event): void {
+  if (event.context.matchedRoute) {
```

Comment:
l: It would be nice to avoid nesting here. Can we just do an early return in this code path?

```js
if (!event.context.matchedRoute) {
  return;
}
```

Ditto with some usages below like the `activeSpan` and `rootSpan` checks below. I'll leave it up to you what balance to strike.